utils/publish-debs.sh

```diff
@@ -8,5 +8,9 @@ cp -a $SECRET_KEY ${keydir}/key
 
 docker run --rm -v `pwd`:/code -v ${keydir}:/keydir calico-build/bionic /bin/sh -c "gpg --import < /keydir/key && debsign -kCalico *_*_source.changes"
 for series in trusty xenial bionic; do
-    docker run --rm -v `pwd`:/code calico-build/${series} /bin/sh -c "dput -u ppa:project-calico/${REPO_NAME} /code/*_*${series}_source.changes"
+    # Use the Distribution header to map changes files to Ubuntu versions, as some of our
+    # packages don't include the Ubuntu version name in the changes file name.
+    for changes_file in `grep -l "Distribution: ${series}" *_source.changes`; do
```

Comment:
Weird, I guess I didn't submit the comment I had written up on this line.
A `for` implies possibly more than one iteration to me but I think only a single 'cycle' at most is expected here. I think I understand why you've used a for here so I was thinking a comment saying that we only expect this to be done once at most would be nice.

Reply:
There will be multiple iterations here.  The `grep ...` will expand to a list of filenames like `felix_3.9.0~0.dev.post47+20190807233357+0000+b144a36~xenial_source.changes nettle_3.3-1_source.changes networking-calico_3.8.0~xenial_source.changes dnsmasq_2.79test1calico1-2~xenial_source.changes`.  Then the code inside the for loop runs for each of those filenames in turn.

Reply:
Ok thanks for the follow up.